From 50c07a8f654f1ff8419a299c320551b6f522c9b2 Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sat, 22 Jan 2022 22:27:36 +0100
Subject: [PATCH 1817/1936] Add CI integration for Arch Linux

.github/workflows/ccpp.yml:
Build in an Arch Linux container against system rtosc and install the
resulting software.
Disable the PortChecker test using sed, as it does not seem to work in
a container environment in github actions:
```
Process not started
/__w/zynaddsubfx/zynaddsubfx/build/src/Tests/check-ports.rb
[no such file or directory]
```
---
 .github/workflows/ccpp.yml | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/.github/workflows/ccpp.yml b/.github/workflows/ccpp.yml
index 700235fa..60cce343 100644
--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -25,3 +25,27 @@ jobs:
       run: cd build && make
     - name: make test
       run: cd build && make test
+  build_arch_linux_system_rtosc:
+    runs-on: ubuntu-latest
+    container:
+      image: archlinux:latest
+    steps:
+      - name: Install dependencies
+        run: pacman --noconfirm -Syu alsa-lib base-devel cmake cxxtest dssi fftw fltk git jack ladspa liblo libxpm mxml portaudio rtosc zlib
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      # test does not seem to work in a container and reports:
+      # Process not started
+      #  /__w/zynaddsubfx/zynaddsubfx/build/src/Tests/check-ports.rb
+      # [no such file or directory]
+      - name: Disable PortChecker test
+        run: sed -e '/add_test(PortChecker/d' -i src/Tests/CMakeLists.txt
+      - name: Build zynaddsubfx
+        run: |
+          cmake -DCMAKE_INSTALL_PREFIX=/usr -DZYN_SYSTEM_RTOSC=ON -B build -S .
+          make VERBOSE=1 -C build
+      - name: Test zynaddsubfx
+        run: make VERBOSE=1 -k test ARGS='-VVd' -C build
+      - name: Install zynaddsubfx
+        run: make VERBOSE=1 install -C build
-- 
2.47.0

